### PR TITLE
BS-14512 create Rest API I18nTranslationV2

### PR DESCRIPTION
**Pull request by julienmege**
BS-14512 clean new api i18nTranslation and add test.

Replace old i18nTranslation API,
by a restlet api, to avoid bad performance imply by the unused parsing of the old api.
by the way, fix the issue, that escape html in translations.

see also the following PR:
https://github.com/bonitasoft/bonita-web-sp/pull/44

### CODE REVIEW

File: server/src/main/webapp/WEB-INF/urlrewrite.xml [hunk 1]
```diff
@@ -226,10 +226,6 @@
         <to>/APIToolkit/system/i18nlocale$2</to>
     </rule>
     <rule>
-        <from>^(/portal/custom-page)?/API/system/i18ntranslation([\?/]|$)</from>
-        <to>/APIToolkit/system/i18ntranslation$2</to>
-    </rule>
```
Comment (cpuy) [inline server/src/main/webapp/WEB-INF/urlrewrite.xml]: :+1:

File: server/src/test/java/org/bonitasoft/web/rest/server/BonitaRestletApplicationTest.java [hunk 2]
```diff
@@ -40,13 +41,26 @@ public void should_application_register_bdm_resources() throws Exception {
...
-        bonitaSPRestletApplication.buildRouter();
+        bonitaRestletApplication.buildRouter();
 
         //then
         Mockito.verify(finderFactory).createExtensionResource();
     }
 
+
+    @Test
+    public void should_application_register_i18n_resources() throws Exception {
```
Comment (cpuy) [inline server/src/test/java/org/bonitasoft/web/rest/server/BonitaRestletApplicationTest.java]: "Should application register i18n resources" is not a well formed english sentence

File: server/src/test/java/org/bonitasoft/web/rest/server/api/system/I18nTranslationResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+        assertThat(response).hasStatus(Status.SUCCESS_OK);
+
+        assertJsonEquals("[" +
+                        "{\"key\": \"key1\", \"value\": \"<strong>message 1</strong>\"}," +
+                        "{\"key\": \"key2\", \"value\": \"autre méssage\"}," +
+                        "{\"key\": \"key3\", \"value\": \"~%^*µ\"}" +
+                        "]",
+                response.getEntityAsText(),
+                JsonAssert.when(Option.IGNORING_ARRAY_ORDER));
```
Comment (cpuy) [inline server/src/test/java/org/bonitasoft/web/rest/server/api/system/I18nTranslationResourceTest.java]: :+1:  Nice !

File: server/src/test/java/org/bonitasoft/web/rest/server/api/system/I18nTranslationResourceTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    @Test
+    public void should_return_http400_error_code_when_no_queryString() throws Exception {
+
+        Response response = request("/system/i18ntranslationv2").get();
```
Comment (cpuy) [inline server/src/test/java/org/bonitasoft/web/rest/server/api/system/I18nTranslationResourceTest.java]: :-1:    `i18ntranslation` 

`/system/i18ntranslationv2` is not mapped to nothing

File: server/src/test/java/org/bonitasoft/web/rest/server/api/system/I18nTranslationResourceTest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    @Test
+    public void should_return_translation_for_the_given_local() throws Exception {
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put("key1", "<strong>message 1</strong>");
+        parameters.put("key2", "autre méssage");
+        parameters.put("key3", "~%^*µ");
+
+        when(i18n.getLocale(AbstractI18n.LOCALE.fr)).thenReturn(parameters);
+
+        Response response = request("/system/i18ntranslationv2?f=locale%3Dfr").get();
```
Comment (cpuy) [inline server/src/test/java/org/bonitasoft/web/rest/server/api/system/I18nTranslationResourceTest.java]: :-1:    `i18ntranslation` 

`/system/i18ntranslationv2` is not mapped to nothing

File: server/src/test/java/org/bonitasoft/web/rest/server/api/system/I18nTranslationResourceTest.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        Response response = request("/system/i18ntranslationv2").get();
+
+        assertThat(response).hasStatus(Status.CLIENT_ERROR_BAD_REQUEST);
+
+    }
+
+    @Test
+    public void should_return_http400_error_code_when_no_locale_param() throws Exception {
+
+        Response response = request("/system/i18ntranslationv2?f=test").get();
```
Comment (cpuy) [inline server/src/test/java/org/bonitasoft/web/rest/server/api/system/I18nTranslationResourceTest.java]: :-1:    `i18ntranslation` 

`/system/i18ntranslationv2` is not mapped to nothing

File: server/src/main/java/org/bonitasoft/web/rest/server/api/system/I18nTanslationResource.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+    public I18nTanslationResource(I18n i18n) {
+        this.i18n = i18n;
+    }
+
+    @Get("json")
+    public List<Translation> getI18nTanslation() throws CommandNotFoundException, CommandExecutionException, CommandParameterizationException {
+        List<Translation> items = new ArrayList<>();
+
+        String local = getSearchFilters()!=null? getSearchFilters().get("locale"):null;
```
Comment (cpuy) [inline server/src/main/java/org/bonitasoft/web/rest/server/api/system/I18nTanslationResource.java]: IMHO we should extract this ugly code in a dedicated private method `getLocale()`

`local` variable should be named `locale`

File: server/src/main/java/org/bonitasoft/web/rest/server/api/system/I18nTanslationResource.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        if (local == null) {
+            getResponse().setStatus(Status.CLIENT_ERROR_BAD_REQUEST, "Request should contains 'locale' parameter.");
+            return null;
```
Comment (cpuy) [inline server/src/main/java/org/bonitasoft/web/rest/server/api/system/I18nTanslationResource.java]: Weird this `return null` isn't it ?

**Comment by cpuy** [comment]
Nice PR !

I doubt that tests pass... ;-)

File: server/src/main/java/org/bonitasoft/web/rest/server/api/system/I18nTanslationResource.java (hunk 8 above)
Comment (julienmege) [inline server/src/main/java/org/bonitasoft/web/rest/server/api/system/I18nTanslationResource.java]: i use this code 
http://www.programcreek.com/java-api-examples/index.php?api=org.restlet.data.Status
as example.

but maybe we can return items  (empty list) ?

**Comment by cpuy** [comment]
:+1: Would you mind squashing your commits ?

**Comment by julienmege** [comment]
squash done!

**Pull request merged**